src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,108 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Ui {
+    private Storage storage;
+    private TaskList taskList;
+    boolean isExit = false;
+    private final Scanner scanner = new Scanner(System.in);
+    public Ui(Storage storage, TaskList taskList) {
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+    private final Scanner scanner = new Scanner(System.in);
+    public void showLine() {
+        System.out.println("______________________________________________");
+    }
+    public void greet() {
+        System.out.println("Hello! I'm duke.Duke\n" +
+                "What can I do for you?");
+    }
+    public String readLine() {
+        return scanner.nextLine();
+    }
+    public void handleInput(String input) {
```

Comment:
Would it be better to extract some logic in this huge function to smaller functions? Maybe it would be more readable and easy to understand by others. :)